addons/mail/models/mail_message.py

```diff
@@ -892,7 +892,7 @@ def _message_format(self, fnames, format_reply=True, legacy=False):
         return vals_list
 
     @api.model
-    def _message_fetch(self, domain, max_id=None, min_id=None, limit=30):
+    def _message_fetch(self, domain, max_id=None, min_id=None, limit=30, has_to_mark_as_read=False):
```

Comment:
This method does too many things, and the signature becomes too complex.

It would be best to define a smaller function instead, e.g. `_message_get`, that just return `messages`. Formatting and `set_message_done` aspects should be done in the context of method callers instead.

This would simplify the code, and make it clearer the subtleties in the different flows:
- `_message_get` in `mailbox` no marking as read, because obviously `inbox`, `starred` should preserve unread status
- manual fetch of messages should mark as read, as long as this feature is supported (i.e. authenticated and non-public).